### PR TITLE
Add examples to XML params

### LINKED ISSUE #1628: Add example to XML param
OpenApi 3.0 supports adding examples to parameters.

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#exampleObject

I propose adding support for this via the param XML comments:
```csharp
/// <param name="keywords" example="hello">A list of search terms</param>
[HttpGet(Name = "SearchProducts")]
public IEnumerable<Product> Get([FromQuery(Name = "kw")]string keywords = "foobar")
```
This populates the swagger-ui form:
image

I can submit a PR for this.

Related issue: https://github.com/mattfrear/Swashbuckle.AspNetCore.Filters/issues/66#issuecomment-615450083

### PR DESCRIPTION
**Pull request by mattfrear**
Fixes #1628.
I ensure the example type is the correct type and not always a string.
Includes unit tests and updated Readme.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,4 +11,5 @@ artifacts/
 .DS_Store
 Thumbs.db
 test/WebSites/CliExample/wwwroot/api-docs/v1/*.json
-test/WebSites/CliExampleWithFactory/wwwroot/api-docs/v1/*.json
\ No newline at end of file
+test/WebSites/CliExampleWithFactory/wwwroot/api-docs/v1/*.json
+*ncrunch*
```
Comment (mattfrear) [inline .gitignore:15]: I use the NCrunch test runner so I added this. I can remove it if you prefer.

**Comment by domaindrivendev** [comment]
👍 @mattfrear

**Comment by Ben230** [comment]
Hi @mattfrear, when do you think it would be possible to accept other datatypes e.g. guids rather than just primitive data types? Love the work you are doing btw!

**Comment by foriequal0** [comment]
Hi. Would you mind if I ask why did you put examples under the parameter's schema? I think it would've generated a more concise document if you put it `parameter.Example` in `XmlCommentsParameterFilter`.

**Comment by mattfrear** [comment]
> Hi. Would you mind if I ask why did you put examples under the parameter's schema? I think it would've generated a more concise document if you put it `parameter.Example` in `XmlCommentsParameterFilter`.

That's a good question. I didn't know that XmlCommentsParameterFilter exists. I added it to the schema.Example because I was copying the existing logic which sets schema examples from XML comments. 
I think you are correct - the XmlCommentsParameterFilter is the more correct place to set this.

**Comment by foriequal0** [comment]
@mattfrear Thank you for the clarification 👍

**Comment by gohawks3** [comment]
Hi Matt, love your work! would you please add an example for IList param type? Is it supported?

**Comment by mattfrear** [comment]
> 
> 
> Hi Matt, love your work! would you please add an example for IList param type? Is it supported?

Yes @gohawks3, lists are supported, see this comment:

https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/1676#issuecomment-760508083

**Pull request merged**